src/test/java/net/kaczmarzyk/spring/data/jpa/utils/JsonUtilsTest.java

```diff
@@ -0,0 +1,123 @@
+ * limitations under the License.
+ */
+package net.kaczmarzyk.spring.data.jpa.utils;
+
+import com.google.gson.JsonParseException;
+import org.assertj.core.api.Assertions;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.Collection;
+
+public class JsonUtilsTest {
+
+    @Rule
+    public ExpectedException exceptionRule = ExpectedException.none();
+
+    @Test
+    public void getSingleValueByNotCompositeKeyWhenValueIsString() {
+        String json = "{ \"key\": \"value\" }";
+        Collection<String> result = JsonUtils.getValuesFromJson(json, "key");
+
+        Assertions.assertThat(result).containsExactly("value");
+    }
+
+    @Test
+    public void getSingleValueByNotCompositeKeyWhenValueIsNumber() {
+        String json = "{ \"key\": 2 }";
+
+        Collection<String> result = JsonUtils.getValuesFromJson(json, "key");
+
+        Assertions.assertThat(result).containsExactly("2");
+    }
+
+    @Test
+    public void getSingleValueByNotCompositeKeyWhenValueIsBoolean() {
+        String json = "{ \"key\": true }";
+
+        Collection<String> result = JsonUtils.getValuesFromJson(json, "key");
+
+        Assertions.assertThat(result).containsExactly("true");
+    }
+
+    @Test
+    public void getSingleValueByNotCompositeKeyWhenValueIsNull() {
+        String json = "{ \"key\": null }";
+
+        Collection<String> result = JsonUtils.getValuesFromJson(json, "key");
+
+        Assert.assertTrue(result.isEmpty());
+    }
+
+    @Test
+    public void getSingleValueWhenValueIsObject() {
+        String json = "{ \"key\": { \"innerKey\": \"value\"}}";
+
+        exceptionRule.expect(JsonParseException.class);
+        exceptionRule.expectMessage("Value by key should be primitive or array primitives");
+
+        JsonUtils.getValuesFromJson(json, "key");
+    }
+
+    @Test
+    public void getMultipleValuesByNotCompositeKey() {
+        String json = "{ \"array\": [\"value1\", \"value2\"] }";
+
+        Collection<String> result = JsonUtils.getValuesFromJson(json, "array");
+
+        Assertions.assertThat(result).containsExactly("value1", "value2");
+    }
+
+    @Test
+    public void getSingleValueByCompositeKey() {
+        String json = "{ \"key1\": { \"key2\": \"value\" }}";
+
+        Collection<String> result = JsonUtils.getValuesFromJson(json, "key1.key2");
+
+        Assertions.assertThat(result).containsExactly("value");
+    }
+
+    @Test
+    public void getValueFromJsonWhenKeyNotPresent() {
+        String json = "{ \"key1\": { \"key2\": \"value\" }}";
+
+        Collection<String> result = JsonUtils.getValuesFromJson(json, "key1.innerKey");
+
+        Assert.assertTrue(result.isEmpty());
+    }
+
+    @Test
+    public void getValueFromJsonWhenMiddleNodeIsArray() {
+        String json = "{ \"key\": { \"array\": [{ \"object1\": \"value1\" }, { \"object2\": \"value2\" }]}}";
```

Comment:
In the documentation you wrote `<b>!!!ATTENTION:</b> Array cannot be root or middle json node.`. Please consider adding example from this test to readme.